chasm/lib/callback/executors_test.go

```diff
@@ -0,0 +1,542 @@
+			expectChasm: true,
+			setupCallback: func(cb *Callback) {
+				cb.Callback.Variant = &callbackspb.Callback_Nexus_{
+					Nexus: &callbackspb.Callback_Nexus{
+						Url: chasm.NexusCompletionHandlerURL,
+					},
+				}
+			},
+		},
+		{
+			name:        "external-nexus-callback",
+			nexusURL:    "http://external:8080",
+			expectNexus: true,
+			setupCallback: func(cb *Callback) {
+				cb.Callback.Variant = &callbackspb.Callback_Nexus_{
+					Nexus: &callbackspb.Callback_Nexus{
+						Url: "http://external:8080",
+					},
+				}
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			// Setup time source
+			timeSource := clock.NewEventTimeSource()
+			timeSource.Update(time.Now())
+
+			// Create callback
+			callback := &Callback{
+				CallbackState: &callbackspb.CallbackState{
+					RequestId: "request-id",
+					Callback: &callbackspb.Callback{
+						Variant: &callbackspb.Callback_Nexus_{
+							Nexus: &callbackspb.Callback_Nexus{
+								Url: tc.nexusURL,
+							},
+						},
+					},
+					Attempt: 1,
+				},
+			}
+			tc.setupCallback(callback)
+
+			// Setup completion getter
+			completion, err := nexusrpc.NewOperationCompletionSuccessful(nil, nexusrpc.OperationCompletionSuccessfulOptions{})
+			require.NoError(t, err)
+			completionGetter := &mockNexusCompletionGetter{
+				completion: completion,
+			}
+			// Set the CanGetNexusCompletion field using reflection
+			setFieldValue(&callback.CompletionSource, CompletionSource(completionGetter))
+
+			// Create mock engine and setup expectations
+			mockEngine := chasm.NewMockEngine(ctrl)
+			mockEngine.EXPECT().ReadComponent(
+				gomock.Any(),
+				gomock.Any(),
+				gomock.Any(),
+			).DoAndReturn(func(ctx context.Context, ref chasm.ComponentRef, readFn func(chasm.Context, chasm.Component) error, opts ...chasm.TransitionOption) error {
+				// Create a mock context
+				mockCtx := &chasm.MockContext{
+					HandleNow: func(component chasm.Component) time.Time {
+						return timeSource.Now()
+					},
+					HandleRef: func(component chasm.Component) ([]byte, error) {
+						return []byte{}, nil
+					},
+				}
+
+				// Call the readFn with our callback
+				return readFn(mockCtx, callback)
+			})
+
+			// Create ComponentRef
+			ref := chasm.NewComponentRef[*Callback](chasm.EntityKey{
+				NamespaceID: "namespace-id",
+				BusinessID:  "workflow-id",
+				EntityID:    "run-id",
+			})
+
+			// Create context with engine
+			ctx := chasm.NewEngineContext(context.Background(), mockEngine)
+
+			// Test loadInvocationArgs via ReadComponent
+			invokable, err := chasm.ReadComponent(
+				ctx,
+				ref,
+				(*Callback).loadInvocationArgs,
+				ctx,
+			)
+
+			// Verify the invokable was created successfully
+			require.NoError(t, err)
+			require.NotNil(t, invokable)
+		})
+	}
+}
+
+// Test saveResult transitions
+func TestSaveResult(t *testing.T) {
```

Comment:
This is redundant IMHO, just test the entire executor.

Reply:
Removed